### PR TITLE
fix: error mismatch

### PR DESCRIPTION
**Pull request by Crayon-Shin-chan-1024**
because of recently modify for storage backend, it breaks the error chain

### CODE REVIEW

**Comment by Crayon-Shin-chan-1024** [comment]
> Can you please run `cargo +nightly fmt --all`

i have run the command, i don't know why the rustfmt seems strange

File: cli/src/args.rs [hunk 1]
```diff
@@ -104,20 +105,30 @@ impl RgbArgs {
         }
 
         let provider = FsBinStore::new(stock_path.clone())?;
-        let mut stock = Stock::load(provider, true).map_err(WalletError::WalletPersist).or_else(|err| {
-            if matches!(err, WalletError::Deserialize(DeserializeError::Decode(DecodeError::Io(ref err))) if err.kind() == ErrorKind::NotFound) {
-                if self.verbose > 1 {
-                    eprint!("stock file is absent, creating a new one ... ");
+        let mut stock = Stock::load(provider, true)
+            .map_err(WalletError::WalletPersist)
```
Comment (dr-orlovsky) [inline cli/src/args.rs:109]: I think we can remove this `map_err` here and move it down, saving some clarity in the checks within `or_else`
```suggestion
```

File: cli/src/args.rs [hunk 2]
```diff
@@ -104,20 +105,30 @@ impl RgbArgs {
...
-                let mut stock = Stock::in_memory();
-                stock.make_persistent(provider, true).map_err(WalletError::StockPersist)?;
-                return Ok(stock);
-            }
-            eprintln!("stock file is damaged, failing");
-            Err(err)
-        })?;
+                eprintln!("stock file is damaged, failing");
+                Err(err)
+            })?;
```
Comment (dr-orlovsky) [inline cli/src/args.rs:131]:
```suggestion
            })
            .map_err(WalletError::WalletPersist)?;
```

File: cli/src/args.rs [hunk 3]
```diff
@@ -104,20 +105,30 @@ impl RgbArgs {
...
-                if self.verbose > 1 {
-                    eprint!("stock file is absent, creating a new one ... ");
+        let mut stock = Stock::load(provider, true)
+            .map_err(WalletError::WalletPersist)
+            .or_else(|err| {
+                if let WalletError::WalletPersist(PersistenceError(ref err)) = err {
+                    if let Some(DeserializeError::Decode(DecodeError::Io(io_err))) =
+                        err.downcast_ref::<DeserializeError>()
+                    {
+                        if io_err.kind() == ErrorKind::NotFound {
```
Comment (dr-orlovsky) [inline cli/src/args.rs:115]: I do not think we need that deep construction; we can do the same way we did before with errors in errors

```suggestion
                let Some(DeserializeError::Decode(DecodeError::Io(io_err))) = err.0.downcast_ref::<DeserializeError>() else {
                    eprintln!("stock file is damaged, failing");
                    return Err(err)
                };
                if io_err.kind() == ErrorKind::NotFound {
```

File: cli/src/args.rs [hunk 4]
```diff
@@ -104,20 +105,30 @@ impl RgbArgs {
...
-                let provider = FsBinStore::new(stock_path)?;
-                let mut stock = Stock::in_memory();
-                stock.make_persistent(provider, true).map_err(WalletError::StockPersist)?;
-                return Ok(stock);
-            }
-            eprintln!("stock file is damaged, failing");
-            Err(err)
-        })?;
+                eprintln!("stock file is damaged, failing");
+                Err(err)
```
Comment (dr-orlovsky) [inline cli/src/args.rs:130]: We would not need this part
```suggestion
                eprintln!("stock file is damaged, failing");
                Err(err)
```

**Review by dr-orlovsky — CHANGES_REQUESTED** [review]

**Pull request closed without merging**